### ISSUE TITLE
Allow to pass several Dockerfiles with -f

### ISSUE THREAD

**Issue by Lucas-C** [projectatomic/dockerfile_lint#86]
Hi.

In order to better integrate this hook as a [pre-commit hook](https://github.com/Lucas-C/pre-commit-hooks-nodejs), it would be very handy to allow for several file names to be passed as argument to the CLI.
A line indicating the file name analysed could be printed before each report, so the user does not get mixed up.

This does not seem very hard to implement, I can submit a PR if you agree with this idea.

The CLI command could then be callable like this: `dockerfile_lint -f component1/Dockerfile component2/Dockerfile`

**Comment by lphiri** [comment]
@Lucas-C  sounds like a reasonable request. Go ahead and submit a PR.

**Comment by dav1x** [comment]
+1 @Lucas-C

**Comment by Lucas-C** [comment]
Damn, due to a limitation of commander.js, this won't be as easy as I expected:
https://github.com/tj/commander.js/issues/690

It would be a significant CLI API change, but the simplest would be to remove the `--dockerfile` option, and use instead the remaining `commandline.args` passed to the program.
Or wait for a fix in commander.js. Or switch to another CLI args parser.

What do you think ?

**Comment by lphiri** [comment]
You should be able to use a list  like so   -f   component1/Dockerfile,component2/Dockerfile
function list(val) {
  return val.split(',');
}

with : .option('-l, --list <items>', 'A list', list)

**Comment by Lucas-C** [comment]
Yes, `pre-commit` provides a space-separated list of files to the hook underlying command, not a comma-separated-one :(
And a comma-separated list of files is very non standard in terms of unix-like CLI programs

**Comment by dav1x** [comment]
Can you not split on the space instead?  The resulting array should still be usable, yes?
```
function list(val) { return val.split(' '); }

```

**Comment by lphiri** [comment]
You can split on space, though you would have to put the list in quotes:
`dockerfile_lint -f "component1/Dockerfile component2/Dockerfile"`

**Comment by Lucas-C** [comment]
Exactly, because the commander.js parser already split args on space characters.
But again, this won't fit as a `pre-commit` hook : it cannot add enclosing quotes. And it not unix-standard usage neither.
I could not find an explicit reference to this in the [POSIX utility standards](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html) But together with the [GNU standards](https://www.gnu.org/prep/standards/standards.html#Command_002dLine-Interfaces) (which mention `getopt_long`) they implicit describe space-separated filenames as standard operands.

**Comment by Lucas-C** [comment]
I submitted a PR that implement this while conserving the `-f flag`.

Usage example:
```
node ./bin/dockerfile_lint -f Dockerfile -f Dockerfile2
```

What do you think ?

**Comment by lphiri** [comment]
Looks like there are some test failures in travis - can we fix those?  Otherwise LGTM

**Comment by Lucas-C** [comment]
Fixed.

Note that I had ro replace the `downloadDockerfile` function by `getContent` which now returns a Promise, because we need to wait for `protocol.get` to complete.

As a consequencewow the linting takes place in parallel on all files.

**Comment by Lucas-C** [comment]
There is still an issue in Travis: because this project should be compatible with NodeJS v0.10/0.11/0.12, `Promise` cannot be used :(

I can try to convert everything back to callbacks, but it's going to be messy without `Promise.all` to wait for their completion.

What do you think ?

**Comment by lphiri** [comment]
I am ok with dropping support for the order nodejs versions . Whats the oldest node version with promise support?

**Comment by Lucas-C** [comment]
From http://node.green I'd say since v0.12.18

**Comment by sky-code** [comment]
Can anyone add a feature for recursion search all `Dockerfile`s from a directory?

**Comment by lphiri** [comment]
@Lucas-C I propose we support v4.0 + to line up with npm.  I merge your PR if you update the travis file.

**Comment by lphiri** [comment]
@sky-code Can you open a separate issue for the dir support?

**Comment by sky-code** [comment]
Yep. #89

**Comment by Lucas-C** [comment]
@lphiri done
